ElectronClient/app/gui/NoteContentPropertiesDialog.tsx

```diff
@@ -1,94 +1,83 @@
 import * as React from 'react';
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 const { _ } = require('lib/locale.js');
-const { themeStyle, buildStyle } = require('../theme.js');
+const { themeStyle } = require('../theme.js');
 const DialogButtonRow = require('./DialogButtonRow.min');
 const Countable = require('countable');
 
 interface NoteContentPropertiesDialogProps {
 	theme: number,
-	text: string,
+	text: number,
+	lines: number,
 	onClose: Function,
 }
 
 interface TextPropertiesMap {
-	[key: string]: Number;
+	[key: string]: number;
 }
 
 interface KeyToLabelMap {
-	[key: string]: String;
+	[key: string]: string;
 }
 
 export default function NoteContentPropertiesDialog(props:NoteContentPropertiesDialogProps) {
-	const styles = buildStyle('NoteContentPropertiesDialog', props.theme, (theme:any) => {
-		return {
-			controlBox: {
-				marginBottom: '1em',
-				display: 'flex',
-				flexDirection: 'row',
-				alignItems: 'center',
-			},
-			button: {
-				minWidth: theme.buttonMinWidth,
-				minHeight: theme.buttonMinHeight,
-				marginLeft: 5,
-				color: theme.color,
-				backgroundColor: theme.backgroundColor,
-				border: '1px solid',
-				borderColor: theme.dividerColor,
-			},
-		};
-	});
+	const theme = themeStyle(props.theme);
+	const textComps: JSX.Element[] = [];
+	const [lines, setLines] = useState<number>(0);
+	const [words, setWords] = useState<number>(0);
+	const [characters, setCharacters] = useState<number>(0);
+	const [charactersNoSpace, setCharactersNoSpace] = useState<number>(0);
+
+	useEffect(() => {
+		Countable.count(props.text, (counter: { words: number; all: number; characters: number; }) => {
+			setWords(counter.words);
+			setCharacters(counter.all);
+			setCharactersNoSpace(counter.characters);
+		});
+		setLines(props.lines);
+	}, []);
+
+	const textProperties: TextPropertiesMap = {
+		lines: lines,
+		words: words,
+		characters: characters,
+		charactersNoSpace: charactersNoSpace,
+	};
```

Comment:
Now that you've updated the code, this object is a bit redundant. You could display directly the words, lines, etc. you defined above. If you need a structure to iterate over, you could simple do `for (const n of ['lines', 'words'...]`.

Reply:
In this case, I kept it in order to continue using the keyToLabel object, rather than having to create some logic to loop through in order to match the right key with the right value.

Reply:
That's reasonable, please leave it as it is then.